openwisp_users/api/mixins.py

```diff
@@ -131,7 +131,11 @@ def _user_attr(self):
         raise NotImplementedError()
 
     def filter_fields(self):
-        user = self.context['request'].user
+        try:
+            user = self.context['request'].user
+        except KeyError:
+            return
+
         if user.is_superuser:
             return
```

Comment:
Possible improvement:
```
user = self.context.get('request', {}).get('user', None)
if !user or user.is_superuser:
   return
```

Will this work?

Reply:
try/except is more suited to exceptional cases

Reply:
> Possible improvement:
> 
> ```
> user = self.context.get('request', {}).get('user', None)
> if !user or user.is_superuser:
>    return
> ```
> 
> Will this work?

@atb00ker I tired implementing it I get this error `AttributeError: 'Request' object has no attribute 'get'`

Reply:
Try this:
```
request = self.context['request']
user request.get('user', None)
```
If this doesn't work, I can try doing it on my machine, but I would recommend to manipulate it to create a solution like this yourself! :smile:

Reply:
@atb00ker Will this be a good replacement??

``` diff
diff --git a/openwisp_users/api/mixins.py b/openwisp_users/api/mixins.py
index 2bfbda8..524649b 100644
--- a/openwisp_users/api/mixins.py
+++ b/openwisp_users/api/mixins.py
@@ -131,9 +131,9 @@ class FilterSerializerByOrganization(OrgLookup):
         raise NotImplementedError()
 
     def filter_fields(self):
-        try:
+        if self.context:
             user = self.context['request'].user
-        except KeyError:
+        else:
             return
```
I tried raising an exception in the else block but, since the error is raised only when the server is started and not when it is running, so it didn't work.

Reply:
Yes, I think that should do the trick as well! :smile: